### PR TITLE
Use controlled vocabulary for species field

### PR DESCRIPTION
**Pull request by nathanathan**
Pivotal Story: https://www.pivotaltracker.com/story/show/144443667

To run the project after this update you will need to extract the ITIS data in your root directory like so:
```
wget https://s3.amazonaws.com/bsve-integration/itisSqlite.zip
unzip itisSqlite.zip
mv itisSqlite* itisSqlite
```

### CODE REVIEW

**Comment by jgoley** [comment]
When I run `unzip itisSqlite.zip` after downloading I get this:
```
End-of-central-directory signature not found.  Either this file is not a zipfile,
or it constitutes one disk of a multi-part archive.  In the latter case the central
directory and zipfile comment will be found on the last disk(s) of this archive.
unzip:  cannot find zipfile directory in one of itisSqlite.zip or itisSqlite.zip.zip, 
and cannot find itisSqlite.zip.ZIP, period.
```
And `Error 2 No such file or directory` when opening it in Finder.

**Comment by nathanathan** [comment]
@jgoley Sorry about that, I uploaded an incomplete file. It should work now.

File: client/controllers/incidentReports/incidentForm.coffee [hunk 1]
```diff
@@ -124,6 +127,31 @@ Template.incidentForm.helpers
...
 
+  speciesOptionsFn: ->
+    return (params, callback) ->
+      term = params.term?.trim()
+      if not term
+        return callback(results: [])
+      Meteor.call 'searchSpeciesNames', term, (error, results)->
+        if error
+          notify('error', error.reason)
+        console.log results
```
Comment (jgoley) [inline client/controllers/incidentReports/incidentForm.coffee]: Should we remove this log?

File: server/updaters.coffee [hunk 2]
```diff
@@ -10,7 +10,7 @@ feedSchema = require '/imports/schemas/feed'
 Constants = require '/imports/constants.coffee'
 { regexEscape } = require '/imports/utils'
 
-DATA_VERSION = 4
+DATA_VERSION = 7
```
Comment (jgoley) [inline server/updaters.coffee]: Just curious why we are jumping up to 7.

File: client/controllers/incidentReports/incidentForm.coffee [hunk 3]
```diff
@@ -124,6 +127,31 @@ Template.incidentForm.helpers
...
+              id: 'tsn:' + item.tsn
+              text: text
+              item: item
+            }
+          ).concat([
+            id: "userSpecifiedSpecies:#{term}"
+            text: "Other Species: #{term}"
+          ])
+        )
+
```
Comment (jgoley) [inline client/controllers/incidentReports/incidentForm.coffee]: There may be a few areas where we could update the code style to stay consistent with [our guide](https://github.com/ecohealthalliance/tater/wiki/Coffeescript-Style-Guide).
Like line 135 (adding space before `->`) and maybe removing some of the parens?

**Comment by jgoley** [comment]
Looks great with the exception of the minor considerations I've pointed out.
👍

File: server/updaters.coffee (hunk 2 above)
Comment (nathanathan) [inline server/updaters.coffee]: That was to get the updaters to run again without resetting my database.

**Pull request merged**